wait/host.go

```diff
@@ -584,51 +584,69 @@ func (a *HostAwaitility) WaitForTemplateUpdateRequests(namespace string, count i
 }
 
 // NotificationWaitCriterion checks if a Notification meets the given condition
-type NotificationWaitCriterion func(a *HostAwaitility, mur *toolchainv1alpha1.Notification) bool
+type NotificationWaitCriterion func(a *HostAwaitility, mur toolchainv1alpha1.Notification) bool
 
-// WaitForNotification waits until there is a Notification available with the given name and the optional conditions
-func (a *HostAwaitility) WaitForNotification(name string, criteria ...NotificationWaitCriterion) (*toolchainv1alpha1.Notification, error) {
-	var notification *toolchainv1alpha1.Notification
+// WaitForNotifications waits until there is a Notification available with the given name and the optional conditions
+func (a *HostAwaitility) WaitForNotifications(notificationName, notificationType string, criteria ...NotificationWaitCriterion) ([]*toolchainv1alpha1.Notification, error) {
+	var notifications []*toolchainv1alpha1.Notification
 	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
-		obj := &toolchainv1alpha1.Notification{}
-		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Namespace, Name: name}, obj); err != nil {
+		labels := map[string]string{toolchainv1alpha1.NotificationUserNameLabelKey: notificationName, toolchainv1alpha1.NotificationTypeLabelKey: notificationType}
+		opts := client.MatchingLabels(labels)
+		notificationList := &toolchainv1alpha1.NotificationList{}
+		if err :=  a.Client.List(context.TODO(), notificationList, opts);  err != nil {
 			if errors.IsNotFound(err) {
-				a.T.Logf("waiting for availability of notification '%s'", name)
+				a.T.Logf("waiting for availability of notification '%s'", notificationName)
 				return false, nil
 			}
 			return false, err
 		}
-		for _, match := range criteria {
-			if !match(a, obj) {
-				return false, nil
+
+		for _, n := range notificationList.Items {
+			for _, match := range criteria {
+				if match(a, n) {
+					a.T.Logf("found notification '%s'", notificationName)
+					notifications = append(notifications, &n)
+				}
 			}
 		}
-		a.T.Logf("found notification '%s'", name)
-		notification = obj
+
+		if len(notifications) <= 0 {
+			return false, nil
+		}
 		return true, nil
 	})
-	return notification, err
+	return notifications, err
 }
 
 // WaitUntilNotificationDeleted waits until the Notification with the given name is deleted (ie, not found)
```

Comment:
```suggestion
// WaitUntilNotificationDeleted waits until the Notification for the given user is deleted (ie, not found)
```

Reply:
Updated as part of: 1ddb1a3c4a441b524ea8e21f03903f6dce26147e